cli/command/service/update.go

```diff
@@ -164,9 +164,16 @@ func runUpdate(dockerCli *command.DockerCli, flags *pflag.FlagSet, options *serv
 	}
 
 	if flags.Changed("image") {
-		if err := resolveServiceImageDigest(dockerCli, spec); err != nil {
+		if err := resolveServiceImageDigestContentTrust(dockerCli, spec); err != nil {
 			return err
 		}
+		noResolveImage, err := flags.GetBool(flagNoResolveImage)
+		if err != nil {
+			return err
+		}
+		if !noResolveImage {
```

Comment:
@aaronlehmann I followed the pattern used below this for `flagRegistryAuth`, assuming that this is how `pflags` work. I'll change it to this and test.

```
	// only send auth if flag was set
	sendAuth, err := flags.GetBool(flagRegistryAuth)
	if err != nil {
		return err
	}
	if sendAuth {
          ...
```